core/src/text/fontContext.cpp

```diff
@@ -2,8 +2,13 @@
 #define FONTSTASH_IMPLEMENTATION
 #include "fontstash.h"
 
+#define LOGE(fmt, ...) do { logMsg("FontContext/Error: " fmt "\n", ## __VA_ARGS__); } while(0)
+#define LOGW(fmt, ...) do { logMsg("FontContext/Warn: " fmt "\n", ## __VA_ARGS__); } while(0)
```

Comment:
Getting a whole lot of warnings here:
```
warning: token pasting of ',' and __VA_ARGS__ is a GNU extension [-Wgnu-zero-variadic-macro-arguments]
```
Also, can you help me understand what's going on in these lines? What's the ## and why is it in a loop?